cmd/admin-handler-utils.go

```
@@ -90,6 +90,18 @@ func toAdminAPIErr(ctx context.Context, err error) APIError {
 		apiErr = errorCodes.ToAPIErrWithErr(e.Code, e.Cause)
 	default:
 		switch {
+		case errors.Is(err, errDecommissionAlreadyRunning):
+			apiErr = APIError{
+				Code:           "XMinioDecommissionNotAllowed",
```

We could return the same error string as `errDecommissionAlreadyRunning` and `errDecommissionComplete` respectively. Wouldn't that be more user-friendly?

---

No we are sending different error for different states - there is a situation when decommission is still running and while another when its fully complete. 

Both are meant to provide different sets of information.